src/Native/System.Native/pal_networking.cpp

```diff
@@ -2257,7 +2275,7 @@ static Error CloseSocketEventPortInner(int32_t port)
     return err == 0 ? PAL_SUCCESS : ConvertErrorPlatformToPal(errno);
 }
 
-static Error TryChangeSocketEventRegistrationInner(int32_t port, int32_t socket, SocketEvents currentEvents, SocketEvents newEvents, uintptr_t data)
+static Error TryChangeSocketEventRegistrationInner(int32_t port, int32_t socket, SocketEvents currentEvents, SocketEvents newEvents, int64_t data)
```

Comment:
I wasn't able to test on Linux before. Now I have the access. On FreeBSD and OSX it was compiling.
I will add a #if #else guard, without changing the function signature.